src/ethereum/cancun/fork.py

```diff
@@ -411,27 +451,33 @@ def check_transaction(
 
     if isinstance(tx, BlobTransaction):
         if not isinstance(tx.to, Address):
-            raise InvalidBlock
+            raise TransactionTypeContractCreationError(tx)
```

Comment:
@souradeep-das you had a type checking error here (`tx[0]`). Make sure to run `tox` (though until #1278 is merged, `tox -e py3,static,doc` is sufficient) before making your PRs.